playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml

```diff
@@ -229,3 +229,56 @@
   tasks:
   - include: docker/upgrade.yml
     when: l_docker_upgrade is defined and l_docker_upgrade | bool and not openshift.common.is_atomic | bool
+
+- name: Drain and upgrade master nodes
```

Comment:
I would like to do that. There are still a minor issues with hooks. The `openshift_node_upgrade` restarts docker which given the current `openshift_master_upgrade_hook` position needs to be done after the hook is run. E.g. we can duplicate the hook in the `openshift_node_upgrade` role and run it only if the node is the master as well and run the original hook in the serial master upgrade loop only when the master is not a node. It looks like a hack but given the `openshift_node_upgrade` role is atomic from the point of view of the serial master upgrade loop, I don't see any other way around.

At the same time it would make again a sense to promote the master upgrade loop content into a role (or most of it).

Given the discussion we had today I will keep the PR as it is and move the upgrade node into the master upgrade play in another PR.